manim/utils/bezier.py

```diff
@@ -369,3 +372,114 @@ def get_quadratic_approximation_of_cubic(a0, h0, h1, a1):
 
 def is_closed(points: typing.Tuple[np.ndarray, np.ndarray]) -> bool:
     return np.allclose(points[0], points[-1])
+
+
+def bez_params_from_point(
+    point: typing.Iterable[typing.Union[float, int]],
+    control_points: typing.Iterable[typing.Iterable[typing.Union[float, int]]],
+) -> list[float]:
+    """Obtains the parameter corresponding to a given point
+    for a given bezier curve.
+
+    Parameters
+    ----------
+    point (typing.Iterable[float])
+        The Cartesian Coordinates of the point whose parameter
+        should be obtained.
+
+    control_points (typing.Iterable[typing.Iterable[float]])
+        The Cartesian Coordinates of the ordered control
+        points of the bezier curve on which the point may
+        or may not lie.
+
+    Returns
+    -------
+        list[float]
+            List containing possible parameters for the given point on
+            the given bezier curve.
+            This usually only contains one or zero elements, but if the
+            point is, say, at the beginning/end of a closed loop, may return
+            a list with more than 1 value, corresponding to the beginning and
+            end etc. of the loop.
+    """
+    assert all(np.shape(point) == np.shape(c_p) for c_p in control_points)
+
+    roots = []
+    for i in range(len(point)):
+        p = point[i]
+        # For now, we only deal with quadratic or cubic.
+        if len(control_points) == 3:
+            p0, p1, p2 = control_points
+            p0 = p0[i]
+            p1 = p1[i]
+            p2 = p2[i]
+            bezier_polynom = np.polynomial.Polynomial(
+                [p0 - (2 * p1) + p2, 2 * (p1 - p0), p0 - p][::-1]
+            )
+        elif len(control_points) == 4:
+            p0, p1, p2, p3 = control_points
+            p0 = p0[i]
+            p1 = p1[i]
+            p2 = p2[i]
+            p3 = p3[i]
+            bezier_polynom = np.polynomial.Polynomial(
+                [
+                    p3 - 3 * p2 + 3 * p1 - p0,  # t**3
+                    3 * (p0 - (2 * p1) + p2),  # t**2
+                    3 * (p1 - p0),  # t
+                    p0 - p,  # Constant
+                ][::-1]
+            )
+
+        if len(bezier_polynom.roots()) > 0:
+            roots.append(bezier_polynom.roots())
+
+    # Of the roots found, remove all imaginary ones, and
+    # round all roots to the number of decimal places of CLOSED_THRESHOLD
+    # Then, get the common roots for the solved bezier equations in all the
+    # supplied dimensions.
+    roots = [
+        r.real
+        for r in reduce(
+            np.intersect1d,
+            [
+                [
+                    np.around(j, int(np.log10(1 / CLOSED_THRESHOLD))).tolist()
+                    for j in i
+                    if j.imag == 0
+                ]
+                for i in roots
+            ],
+        )
+    ]
```

Comment:
I think the code would be a lot cleaner if this could be broken up into multiple statements instead of one long list comprehension

Reply:
Side Note: I couldn't find any mention of CLOSED_THRESHOLD anywhere else in this repo. Checking 3b1b's repo shows that it was used a few years ago, but isn't used there anymore as well. I think it should be removed, and `VMobject.tolerance_for_point_equality` should be used instead.

Reply:
Done.